cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -361,23 +369,29 @@ private void deleteScheduleInScheduler(ProgramSchedule schedule) throws NotFound
   @Override
   public void deleteSchedules(Iterable<? extends ScheduleId> scheduleIds) throws NotFoundException {
     checkStarted();
-    execute((StoreAndQueueTxRunnable<Void, NotFoundException>) (store, queue) -> {
+    execute((StoreQueueAndProfileTxRunnable<Void, NotFoundException>) (store, queue, profileDataset) -> {
       long deleteTime = System.currentTimeMillis();
       for (ScheduleId scheduleId : scheduleIds) {
         ProgramSchedule schedule = store.getSchedule(scheduleId);
         deleteScheduleInScheduler(schedule);
         queue.markJobsForDeletion(scheduleId, deleteTime);
         adminEventPublisher.publishScheduleDeletion(scheduleId, schedule);
+        // if the deleted schedule has properties with profile assignment, remove the assignment
+        Optional<ProfileId> profileId = SystemArguments.getProfileIdFromArgs(scheduleId.getNamespaceId(),
+                                                                             schedule.getProperties());
+        if (profileId.isPresent()) {
+          profileDataset.removeProfileAssignment(profileId.get(), scheduleId);
+        }
       }
       store.deleteSchedules(scheduleIds);
       return null;
     }, NotFoundException.class);
   }
 
   @Override
-  public void deleteSchedules(ApplicationId appId) {
+  public void deleteSchedules(ApplicationId appId) throws NotFoundException {
```

Comment:
is this thrown if we try to remove the assignment and the profile doesn't exist somehow? In that case I think we should log some internal warning about inconsistent state, but continue on. Otherwise if we ever get into this situation, either through a bug in our code or through data corruption, the user will never be able to delete the schedules.